### DISCUSSION TITLE
How to add Shadcn to exisiting mono-repos?

### DISCUSSION THREAD

**Discussion by alamenai** [shadcn-ui/ui#6225] [Q&A]
I have already a mono repo and I want to start using Shadcn CLI but it is only for fresh projects, is there a way how to add it to an existing mono-repos:

```
 npx shadcn@canary init
✔ Preflight checks.
✖ Verifying framework.

We could not detect a supported framework at /home/aldin-bug/Desktop/one-night.
Visit https://ui.shadcn.com/docs/installation/manual to manually configure your project.
Once configured, you can use the cli to add components.

```

**Comment by flippyhead** [comment]
Yeah I found this confusing too, it doesn't seem to work unless you run this from within the apps directory, in which case it creates a new monorepo structure within your monorepo structure.

**Comment by johanwulf** [comment]
Tried adding it to an existing monorepo, same issues as above.

Tried creating a new monorepo with the command (which is how I suppose it should be done (?))
```
✦ ❯ npx shadcn@canary init
✔ The path /Users/johan.wulf/repos does not contain a package.json file.
  Would you like to start a new project? › Next.js (Monorepo)
✔ What is your project named? … test
✖ Something went wrong creating a new Next.js monorepo.
Something went wrong. Please check the error below for more details.
If the problem persists, please open an issue on GitHub.

Command failed with exit code 1: npm install
npm error code EUNSUPPORTEDPROTOCOL
npm error Unsupported URL Type "workspace:": workspace:*
```

Not sure how it should be used at all 😄

**Comment by Husain010** [comment]
Anyone found a way out for this? Any ways to add it to an existing monorepo? especially in packages/ui or something like that so we can use it in all the apps. @shadcn

**Reply by prashik0202** [reply]
@Husain010 you can checkout this [link](https://matinkhani.medium.com/create-a-turborepo-with-nextjs-tailwindcss-shadcn-6e6ecfd52aea)
